openapi/mojaloop-connector.yaml

```diff
@@ -0,0 +1,281 @@
+externalDocs:
+  description: Find out more about Rafiki
+  url: https://rafiki.dev/
+servers:
+  - url: https://rafiki.connector.mojaloop/
+tags:
+  - name: quote
+    description: quote related operations
+  - name: transfer
+    description: transfer related operations
+paths:
+  /mjl-quote:
+    post:
+      tags:
+        - quote
+      summary: Request a new Mojaloop quote.
+      description: Request a new Mojaloop quote, which will be routed to the CNP (Cross Network Provider) Mojaloop Hub instance.
+      operationId: requestQuote
+      requestBody:
+        description: Create a new quote request.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/mjl-quote-req'
+        required: true
+      parameters:
+        - name: fspiop-source
+          in: header
+          description: 'The DFSP that requested the quote'
+          required: false
+          schema:
+            type: string
+        - name: fspiop-destination
+          in: header
+          description: 'The DFSP destination'
+          required: false
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/mjl-quote-rsp'
+        '400':
+          description: Missing fields or malformed syntax
+        '500':
+          description: Server error
+  /mjl-transfer:
+    post:
+      tags:
+        - transfer
+      summary: Request a new Mojaloop transfer. Even though Mojaloop support 2-phase transfers, the Rafiki Mojaloop Connector will invoke a single request (SMS not DMS).
+      description: Request a new Mojaloop transfer, which will be routed to the CNP (Cross Network Provider) Mojaloop Hub instance.
+      operationId: requestTransfer
+      requestBody:
+        description: Create a new transfer request.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/mjl-transfer-req'
+        required: true
+      parameters:
+        - name: fspiop-source
+          in: header
+          description: 'The DFSP that requested the transfer'
+          required: false
+          schema:
+            type: string
+        - name: fspiop-destination
+          in: header
+          description: 'The DFSP destination'
+          required: false
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/mjl-transfer-rsp'
+        '500':
+          description: Invalid input
+components:
+  schemas:
+    mjl-quote-req:
+      required:
+        - transactionId
+        - quoteId
+        - payer
+        - payee
+        - amountType
+      type: object
+      properties:
+        transactionId:
```

Comment:
To double check, we generate this transactionId, and it's then the same one that we pass into the transfer, correct?

Reply:
Yes, I think the `transactionId` would be generated as early as the "party" lookup phase.